src/lib/src/component/navigation/navigation.component.spec.ts

```diff
@@ -758,4 +759,75 @@ it('emits "navigation.open" message when the enter key is pressed', () => {
   }));
 });
 
+  it('re-retrieves test status when the workspace is refreshed', async(() => {
+    // given
+    setupWorkspace(component, fixture);
+    let reloadedWorkspace = WorkspaceElement.copyOf(component.getWorkspace().root);
+    when(persistenceService.listFiles()).thenReturn(Promise.resolve(reloadedWorkspace));
+    when(executionService.statusAll()).thenReturn(Promise.resolve(
+      new Map<string, ElementState>([[tclFile.path, ElementState.LastRunFailed]])));
+    when(executionService.status(tclFile.path)).thenReturn(Promise.resolve(
+      new Response(new ResponseOptions({ body: 'FAILED' }))));
+
+    // when
+    component.refresh();
+
+    // then
+    fixture.whenStable().then(() => {
+      let updatedTclFile = component.getWorkspace().getElement('subfolder/file.tcl');
```

Comment:
at this point I have no idea that tclFile.path == 'subfolder/file.tcl'. using tclFile.path might be confusing too. Maybe put a 'let updatedTclFilePath = tclFile.path;' in the given section and use that?

Reply:
Absolutely right, I introduced a new variable to make it clearer, but I chose to name it `pathInWorkspaceToBeRefreshed` (because the path remains the same, while the object identities of the corresponding `WorkspaceElement` instances are expected to change; `updatedTclFilePath` kind of carries with it the notion that it is only the path of the element _after_ the refresh).